### PR TITLE
Goroutines not always stopped

### PR DESCRIPTION
**Pull request by caproven**
Input welcome - I don't have much experience with contexts or wait groups.

Contexts & wait groups seem like standard tools for coordinating shutdown, however error propagation up the call stack gets complicated. Imo forcing multiple parts of the codebase to gracefully handle this makes for error-prone code, in addition to bloating func signatures.

For a first pass I've added [errgroup](https://pkg.go.dev/golang.org/x/sync/errgroup) to simplify usage.

In the interest of graceful shutdown, I've also added signal handling. Also made some minor tweaks to tls params

### CODE REVIEW

**Review by xbcsmith — APPROVED** [review]
Want another set of eyes before I merge

**Review by ninjamast3r — APPROVED** [review]
I've not used errgroup before, but it seems cool. I think this will work.

**Pull request merged**
